vumi_twilio_api/tests/test_twilio_api.py

```diff
@@ -21,80 +25,122 @@ class TwiMLServer(object):
 
     def __init__(self, responses={}):
         self._responses = responses.copy()
+        self.requests = []
 
     def add_response(self, filename, response):
         self._responses[filename] = response
 
+    def add_err(self, filename, err):
+        self._responses[filename] = Exception(err)
+
     @app.route('/<string:filename>')
     def get_twiml(self, request, filename):
+        self.requests.append({
+            'filename': filename,
+            'request': request,
+        })
+        response = self._responses[filename]
+        if isinstance(response, Exception):
+            request.setResponseCode(500)
+            return response.message
         request.setHeader('Content-Type', 'application/xml')
-        return ET.tostring(self._responses[filename])
+        return str(self._responses[filename])
+
+    @app.route('/')
+    def get_root(self, request):
+        return self.get_twiml(request, '')
+
+    @inlineCallbacks
+    def start(self):
+        site_factory = LogFilterSite(self.app.resource())
+        self._webserver = yield reactor.listenTCP(
+            0, site_factory, interface='127.0.0.1')
+        self.addr = self._webserver.getHost()
+        self.url = "http://%s:%s/" % (self.addr.host, self.addr.port)
+
+    @inlineCallbacks
+    def stop(self):
+        yield self._webserver.stopListening()
+        yield self._webserver.loseConnection()
```

Comment:
We should probably move the TwiML server out into a `tests/helpers` file and maybe make it a proper `IHelper`.